Lib/xml/etree/ElementTree.py

```diff
@@ -508,6 +511,58 @@ def __eq__(self, other):
 
 # --------------------------------------------------------------------
 
+class XMLDeclarationQuotes(enum.Enum):
+    """
+    Whether or not single quotes or double quotes ought to be used in the XML
+    declaration.
+
+    *SINGLE* (default): <?xml version='1.0' encoding='UTF-8'?>
+    *DOUBLE*: <?xml version="1.0" encoding="UTF-8"?>
+    """
+    SINGLE = "'"
+    DOUBLE = '"'
+
+    def __str__(self):
+        return self.value
+
+class ShortEmptyElements(enum.Enum):
+    """
+    This class creates backwards compatibility with the boolean value of
+    *short_empty_elements* that existed prior to 3.??.
+
+    Assuming the tag `<q/>`, the results will be:
+
+    *SPACE* (default): `<q />`
+    *NOSPACE*: `<q/>`
+    *NONE*: `<q></q>`
+    """
+    SPACE = " "
+    NOSPACE = ""
+    NONE = False
+
+    def __bool__(self):
+        return self != ShortEmptyElements.NONE
+
+    @classmethod
+    def _missing_(cls, value):
+        if value is enum.no_arg:
+            return cls.SPACE
+        elif isinstance(value, bool):
+            return cls.SPACE if value else cls.NONE
+        else:
+            return super()._missing_(value)
+
+    @classmethod
+    def tag_defaultdict(cls, short_empty_elements):
+        if not isinstance(short_empty_elements, collections.defaultdict):
+            if isinstance(short_empty_elements, ShortEmptyElements):
+                return collections.defaultdict(lambda: short_empty_elements)
+            elif bool(short_empty_elements) is True:
```

Comment:
Not sure what the intention is here, but the spelling is quite redundant.
```suggestion
            elif short_empty_elements:
```